From 5bfd773ab0a1cbc8d473f1648b98f587c7c489df Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Tue, 19 Jul 2016 16:58:37 +0200
Subject: [PATCH 2/2] crypt: Remove resize support

Currently resize support for crypt backend is broken, because ssm needs
to resize everything in correct order, however it does not at the
moment. So remove the resize support from crypt backend to avoid further
problems, before the support infrastructure is implemented.

Signed-off-by: Lukas Czerner <lczerner@redhat.com>
---
 ssmlib/backends/crypt.py    | 5 -----
 ssmlib/backends/template.py | 4 ++++
 2 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/ssmlib/backends/crypt.py b/ssmlib/backends/crypt.py
index 679d9c0..5da55db 100644
--- a/ssmlib/backends/crypt.py
+++ b/ssmlib/backends/crypt.py
@@ -225,11 +225,6 @@ class DmCryptVolume(DmObject, template.BackendVolume):
         self.run_cryptsetup(command)
         misc.wipefs(vol['crypt_device'], CRYPT_SIGNATURES)
 
-    def resize(self, dm, size, resize_fs=True):
-        size = str(int(size) * 2)
-        command = ['resize', '--size', size, dm]
-        self.run_cryptsetup(command)
-
 
 class DmCryptDevice(DmObject, template.BackendDevice):
 
diff --git a/ssmlib/backends/template.py b/ssmlib/backends/template.py
index 49c3cc0..56bed56 100644
--- a/ssmlib/backends/template.py
+++ b/ssmlib/backends/template.py
@@ -83,6 +83,10 @@ class BackendVolume(Backend):
         self.problem.check(self.problem.NOT_SUPPORTED,
                         "Removing volume with {0} backend".format(self.type))
 
+    def resize(self, volume, size, resize_fs=True):
+        self.problem.check(self.problem.NOT_SUPPORTED,
+                        "Resizing volume with {0} backend".format(self.type))
+
 
 class BackendDevice(Backend):
     def __init__(self, *args, **kwargs):
-- 
1.8.3.1

